caluma/form/validators.py

```diff
@@ -70,6 +71,27 @@ def _validate_question_multiple_choice(self, question, value, **kwargs):
                 f"Should be one of the options [{', '.join(options)}]"
             )
 
+    def _validate_question_dynamic_choice(self, question, value, **kwargs):
+        options = get_data_source_data(document, question.data_source).values_list(
+            "slug", flat=True
+        )
+        if not isinstance(value, str) or value not in options:
+            raise exceptions.ValidationError(
+                f"Invalid value {value}. "
+                f"Should be of type str and one of the options {'.'.join(options)}"
+            )
+
+    def _validate_question_dynamic_multiple_choice(self, question, value, **kwargs):
+        options = get_data_source_data(
+            kwargs["document"], question.data_source
+        ).values_list("slug", flat=True)
```

Comment:
`get_data_source_data` returns a list of `Data`-objects, not a queryset (as the dynamic options are not in the db). This means `values_list` and all other queryset-features are not available.